src/analytics/publisher.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  DepartureGroupsPayload,
+  DepartureGroupsQuery,
+  DepartureRealtimeQuery,
+  DeparturesFromLocationPagingQuery,
+  FeatureLocation,
+  TripPatternsQuery
+} from '../service/types';
+import { PubSub, Topic } from '@google-cloud/pubsub';
+
+type Topics = {
+  departuresSearchGroupsTopic: string;
+  departuresSearchRealtimeTopic: string;
+  departuresSearchTopic: string;
+  getFeaturesTopic: string;
+  getTripPatternsTopic: string;
+};
+
+type Meta = {
+  environment: string;
+};
+
+export default (pubSub: PubSub, topics: Topics, meta: Meta) => {
+  const departuresSearchGroupsTopic = batchedTopic(
+    pubSub,
+    topics.departuresSearchGroupsTopic
+  );
+  const departuresSearchRealtimeTopic = batchedTopic(
+    pubSub,
+    topics.departuresSearchRealtimeTopic
+  );
+  const departuresSearchTopic = batchedTopic(
+    pubSub,
+    topics.departuresSearchTopic
+  );
+  const getFeaturesTopic = batchedTopic(pubSub, topics.getFeaturesTopic);
+  const getTripPatternsTopic = batchedTopic(
+    pubSub,
+    topics.getTripPatternsTopic
+  );
+
+  return {
+    async departuresSearchGroups(
+      payload: DepartureGroupsPayload,
+      query: DepartureGroupsQuery
+    ) {
+      await departuresSearchGroupsTopic.publish(serialize({ payload, query }), {
```

Comment:
Ikke behov for createTopic lengre?

Reply:
Terraform tar seg av det. Enn så lenge vet vi at de eksisterer :).